ext/SciMLBaseZygoteExt.jl

```diff
@@ -299,4 +300,23 @@ end
     ∇responsible_map(__context__, f, args...)
 end
 
+@_adjoint_keepthunks function Zygote.literal_getfield(x::ODEProblem, ::Val{f}) where f
+  val = getfield(x, f)
+  function back(Δ)
+    # error()
+    Zygote.accum_param(__context__, val, Δ) === nothing && return
+    if isimmutable(x)
+      error()
+      dx = (; Zygote.nt_nothing(x)..., pair(Val(f), Δ, x)...)
+      (_project(x, dx), nothing)
+    else
+      dx = Zygote.grad_mut(__context__, x)
+      dx[] = (; dx[]..., pair(Val(f), Zygote.accum(getfield(dx[], f), Δ))...)
+      return (dx,nothing)
+    end
+  end
+  Zygote.unwrap(val), back
+end
+Zygote.accum(::Tuple{}, ::NamedTuple{}) = ()
```

Comment:
Isn't this type piracy?

Reply:
It is 😅

Reply:
https://github.com/FluxML/Zygote.jl/pull/1574 is definitely the safer bet.

Why are problem types `mutable`? Do they need to be?